types/styled-system/index.d.ts

```diff
@@ -37,9 +37,10 @@ export type ResponsiveValue<
     ThemeType extends Theme = RequiredTheme,
 > = T | null | Array<T | null> | { [key in ThemeValue<"breakpoints", ThemeType> & string | number]?: T };
 
-export type ThemeValue<K extends keyof ThemeType, ThemeType, TVal = any> = ThemeType[K] extends TVal[] ? number
-    : ThemeType[K] extends Record<infer E, TVal> ? E
-    : ThemeType[K] extends ObjectOrArray<infer F> ? F
+export type ThemeValue<K extends keyof ThemeType, ThemeType, TVal = any> = NonNullable<ThemeType[K]> extends TVal[]
```

Comment:
I'm surprised this is necessary. Conditional types are distributive, so if `undefined` does not match any of the `extends` types then it should end up in the `never` branch with the only caveat being how `undefined` is handled when compiling under `"strictNullChecks": false`. Can you add a test showing this change is working as expected?

Reply:
Do you mean adding a testing-only definition of `ThemeValue` without these `NonNullable`s?